File: src/simulation/culture/group/cultureaspect/CultureAspect.java
package simulation.culture.group.cultureaspect;

import simulation.culture.group.CultureCenter;
import simulation.culture.group.Group;
import simulation.culture.group.request.Request;

public interface CultureAspect {
    Request getRequest();
    void use(CultureCenter center);
    CultureAspect copy(Group group);
}

File: src/simulation/culture/group/cultureaspect/CultureAspect.kt
package simulation.culture.group.cultureaspect

import simulation.culture.group.centers.Group
import simulation.culture.group.request.Request

interface  CultureAspect {
    fun getRequest(group: Group): Request?
    fun use(group: Group)
    fun adopt(group: Group): CultureAspect?
    fun die(group: Group)
}